python/cuml/common/array.py

```diff
@@ -168,7 +168,10 @@ def __getitem__(self, slice):
     def __setitem__(self, slice, value):
         cp.asarray(self).__setitem__(slice, value)
 
-    def __reduce__(self):
+    def __reduce_ex__(self, protocol):
+        return self.__class__, (self.to_output('numpy'),)
+
+    def __reduce__(self, protocol):
```

Comment:
Would just stick with `__reduce_ex__` and drop `__reduce__`. Pickle just needs one of them implemented.

Also as a note `__reduce__` doesn’t take any additional arguments.